### ISSUE TITLE
Make LaTeX syntax simpler and more general

### ISSUE THREAD

**Issue by scoskey** [coteditor/CotEditor#1279]
**Is your feature request related to a problem? Please describe.**
I use the LaTeX syntax frequently. Many commands that I use are not highlighted by the syntax highlighter. The difficulty is the extensibility of LaTeX which allows for an infinitude of commands. For example the current syntax does not highlight \newcommand, \vspace{}, \hspace{}, \emph{}, \fill, \itemsep, and on and on. I initially started adding commands to the syntax one at a time, but eventually gave up as it would never end.

There are some other minor issues. For example it highlights \texttt{.+} but should really only be \texttt{.+?} in case two instances appear in the same line. For another example it has \frac{}{} highlighted in a unique way, but what is so special about \frac compared with hundreds of other equation entities?

**Describe the solution you'd like**
The syntax should highlight any \blah, since in TeX, \blah is always a command of some kind. It should additionally highlight parameters inside {...} and optional parameters inside [...] and of course equations.

**Describe alternatives you've considered**
I have created my own syntax definition and it is much simpler and more general than the included one. Here are the relevant parts:
```
commands:
- beginString: '[\\¥][a-zA-Z]+'
  regularExpression: true
types:
- beginString: '['
  endString: ']'
attributes:
- beginString: '{'
  endString: '}'
values:
- beginString: '[\\¥](La)?TeXe?'
  regularExpression: true
- beginString: \$([^$\n]+)\$
  regularExpression: true
- beginString: \(
  endString: \)
- beginString: \[
  endString: \]
```
(Note that I left keywords, variables, numbers, strings, and characters all empty).

**Additional context**
Here is what the highlighting looks like. It seems normal enough to me.
<img width="969" alt="Screen Shot 2022-04-06 at 22 00 51" src="https://user-images.githubusercontent.com/1053991/162117622-0715df20-93ea-4d94-b597-bce0fe0dc6ff.png">

**Comment by 1024jp** [comment]
@scoskey Thank you for the feedback. Based on your suggestion, I completely rewrote the LaTeX syntax style. This change is included in CotEditor 4.2.3, which will be released tomorrow.

**Comment by scoskey** [comment]
It looks excellent, thank you!!
